app/ts/background/popupMessageHandlers.ts

```diff
@@ -272,9 +272,8 @@ export async function refreshPopupConfirmTransactionSimulation(simulator: Simula
 			case 'Transaction': {
 				if (transactionToSimulate.success) {
 					return { ...transactionOrMessage, transactionToSimulate, simulationResults: refreshMessage, transactionOrMessageCreationStatus: 'Simulated' }
-				} else {
-					return { ...transactionOrMessage, transactionToSimulate, simulationResults: refreshMessage, transactionOrMessageCreationStatus: 'FailedToSimulate' }
 				}
+				return { ...transactionOrMessage, transactionToSimulate, simulationResults: refreshMessage, transactionOrMessageCreationStatus: 'FailedToSimulate' }
```

Comment:
Why not instead do this?
```ts
return { ...transactionOrMessage, transactionToSimulate, simulationResults: refreshMessage, transactionOrMessageCreationStatus: transactionToSimulate.success ? 'Simulated' : 'FailedToSimulate' }
```

Reply:
I think the typescript cannot handle it as the types depend on each other

Reply:
Hmm, OK.  In that case I prefer if/else for the same reason as below.  Doesn't really feel like a guard clause, but just a branch.